### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by GlizzyProductions**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -58,46 +60,72 @@ Purpose:  This project will show you the difference between member functions and
...
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float num1 { 0 }, num2 { 0 };
+    float updateNums(float* updated)      //12
     {
+        if(updated != nullptr)
+
```
Comment (matkatmusic) [inline main.cpp]: When you write an `if` statement like this, the line that immediately follows is the only line that is conditionally executed. You must use curly braces if you want to have an entire block of code be executed conditionally.
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -58,46 +60,72 @@ Purpose:  This project will show you the difference between member functions and
...
         
+        return num2 * num1;
     }
 };
 
-struct <#structname2#>
+struct UpdateThat
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateNums(U* that, float updated )        //10
```
Comment (matkatmusic) [inline main.cpp]: The typename of the 2nd argument was `<#type name#>*`
The star `*` means it's a pointer.
Make the 2nd argument a pointer
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -58,46 +60,72 @@ Purpose:  This project will show you the difference between member functions and
...
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(that != nullptr)
```
Comment (matkatmusic) [inline main.cpp]: When you write an `if` statement like this, the line that immediately follows is the only line that is conditionally executed. You must use curly braces if you want to have an entire block of code be executed conditionally.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 4]
```diff
@@ -58,46 +60,72 @@ Purpose:  This project will show you the difference between member functions and
...
+struct UpdateThat
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateNums(U* that, float* updated )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(that != nullptr)
```
Comment (matkatmusic) [inline main.cpp]: What do we always do before using pointers?
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Never forget the most important lesson about pointers.

File: main.cpp (hunk 4 above)
Comment (GlizzyProductions) [reply main.cpp]: Oh shoot!!! you said it about a million times and one still slipped by me

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**